src/sourmash/tax/__main__.py

```diff
@@ -0,0 +1,300 @@
+                                                                                       fail_on_missing_taxonomy=args.fail_on_missing_taxonomy)
+
+    if not gather_results:
+        notify(f'No gather results loaded. Exiting.')
+        sys.exit(-1)
+
+    # actually summarize at rank
+    summarized_gather = {}
+    seen_perfect = set()
+    for rank in sourmash.lca.taxlist(include_strain=False):
+        summarized_gather[rank], seen_perfect = tax_utils.summarize_gather_at(rank, tax_assign, gather_results, skip_idents=idents_missed,
+                                                                split_identifiers=not args.keep_full_identifiers,
+                                                                keep_identifier_versions = args.keep_identifier_versions,
+                                                                seen_perfect = seen_perfect)
+
+    # write summarized output csv
+    if "summary" in args.output_format:
+        summary_outfile = make_outfile(args.output_base, ".summarized.csv")
+        with FileOutputCSV(summary_outfile) as out_fp:
+            tax_utils.write_summary(summarized_gather, out_fp)
+
+    # if lineage summary table
+    if "lineage_summary" in args.output_format:
+        lineage_outfile = make_outfile(args.output_base, ".lineage_summary.tsv")
+
+        ## aggregate by lineage, by query
+        lineageD, query_names, num_queries = tax_utils.aggregate_by_lineage_at_rank(summarized_gather[args.rank], by_query=True)
+
+        with FileOutputCSV(lineage_outfile) as out_fp:
+            tax_utils.write_lineage_sample_frac(query_names, lineageD, out_fp, format_lineage=True, sep='\t')
+
+    # write summarized --> krona output tsv
+    if "krona" in args.output_format:
+        krona_resultslist = tax_utils.format_for_krona(args.rank, summarized_gather)
+
+        krona_outfile = make_outfile(args.output_base, ".krona.tsv")
+        with FileOutputCSV(krona_outfile) as out_fp:
+            tax_utils.write_krona(args.rank, krona_resultslist, out_fp)
+
+
+def classify(args):
+    """
+    taxonomic classification of genomes from gather results
+    """
+    set_quiet(args.quiet)
+
+    # first, load taxonomic_assignments
+    tax_assign = {}
+    available_ranks = set()
+    for tax_csv in args.taxonomy_csv:
+
+        try:
+            this_tax_assign, _, avail_ranks = tax_utils.load_taxonomy_csv(tax_csv, split_identifiers=not args.keep_full_identifiers,
+                                              keep_identifier_versions = args.keep_identifier_versions,
+                                              force=args.force)
+            # to do -- maybe check for overlapping tax assignments? rn later ones will override earlier ones
+            tax_assign.update(this_tax_assign)
+            available_ranks.update(set(avail_ranks))
+        except ValueError as exc:
+            error(str(exc))
```

Comment:
should this exit afterwards? usually when I use `error(...)` it means that (barring a `--force`) we should exit.